bin/microstrax

```diff
@@ -119,6 +124,49 @@ def get_data(
     return [dict(zip(row.dtype.names, row)) for row in x]
 
 
+def now(plus=0):
+    return datetime.now(pytz.utc) + timedelta(seconds=plus)
+
+
+def _set_state():
+    """
+    Inform the bootstrax collection we're running microstrax
+    """
+    # DAQ database
+    daq_db_name = 'daq'
+    daq_db_password = straxen.get_secret('MONGO_DAQ_PASSWORD'.lower())
+    daq_db_username = straxen.get_secret('MONGO_DAQ_USERNAME'.lower())
+    daq_client = pymongo.MongoClient(f"mongodb://{daq_db_username}:{daq_db_password}@xenon1t-daq:27017/admin")
+    daq_db = daq_client[daq_db_name]
+    bs_coll = daq_db['eb_monitor']
+    while True:
```

Comment:
For threads like this that constantly run in the background, I like to add a signal handler so it exits cleanly via ctrl-c.

Reply:
Thanks, actually this is not really needed in python:
https://stackoverflow.com/questions/56144570/how-to-stop-a-python-daemon-thread-with-infinite-loop-at-shutdown

Reply:
Fancy.

I've had less-than-perfect success with `pymongo.errors.NotMasterError` - sometimes it doesn't throw that error, so I just wrapped a generic `try...except` rather than one specifically targeting that one.

Reply:
Good point will track other errors and add those as we go. I think I did the same for the bootstrax --undying mode